### PR TITLE
Optimize Metadata Initialization for PHP [WIP]

### PR DESCRIPTION
**Pull request by TeBoring**

### CODE REVIEW

**Comment by PierrickVoulet** [comment]
@TeBoring This PR seems specific to the C implementation, are you planning to do the same in the PHP implementation?

**Comment by TeBoring** [comment]
Yeah. I’ll add php part later. C part should be testable

**Comment by PierrickVoulet** [comment]
@TeBoring it runs fine on my side with some protos generated with version 3.11.2.

This is the only required runtime edition right? If this is the case, what bout creating a new dedicated branch/PR with all the protoc changes for aggregation purpose?

File: php/ext/google/protobuf/protobuf.c [hunk 1]
```diff
@@ -248,6 +250,23 @@ EnumDescriptorInternal* get_class_enumdesc(const char* klass) {
   }
 }
 
+void add_file(const char* file) {
+  printf("File registered: %s\n", file);
```
Comment (PierrickVoulet) [inline php/ext/google/protobuf/protobuf.c]: @TeBoring printing each registered file could take a significant amount of time, is this really necessary?

**Comment by haberman** [comment]
Closing as obsolete.

**Pull request closed without merging**